vim/core/oni-markdown-preview/src/index.tsx

```diff
@@ -0,0 +1,38 @@
+const path = require("path")
+const os = require("os")
+
+import * as React from "react" 
+
+import { MarkdownPreview } from "./MarkdownPreview"
+
+export class MarkdownPreviewEditor implements Oni.Editor {
+
+    constructor(
+        private _oni: any
```

Comment:
A good next step would be to listen to the buffer update events here - we could do:

```
Oni.editors.activeEditor.onBufferChanged.subscribe((evt) => { console.log("Buffer Changed"; debugger; })
```

That way, if you have the debugger open, you can inspect `evt` and see the values for it. We can get the filename + filetype (language) and decide if we should render a preview for it.

If we should, we can get the content via the `contentChanges` array, and then render it out to our component using `marked`.